x-pack/functionbeat/provider/aws/op_wait_cloud_formation.go

```diff
@@ -0,0 +1,59 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package aws
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+var periodicCheck = 10 * time.Second
+
+type opCloudWaitCloudFormation struct {
+	log       *logp.Logger
+	svc       *cloudformation.CloudFormation
+	stackName string
+}
+
+func newOpWaitCloudFormation(
+	log *logp.Logger,
+	cfg aws.Config,
+	stackName string,
+) *opCloudWaitCloudFormation {
+	return &opCloudWaitCloudFormation{
+		log:       log,
+		svc:       cloudformation.New(cfg),
+		stackName: stackName,
+	}
+}
+
+func (o *opCloudWaitCloudFormation) Execute() error {
+	o.log.Debug("Waiting for cloudformation confirmation")
+	status, reason, err := queryStackStatus(o.svc, o.stackName)
+
+	for strings.Index(string(*status), "FAILED") == -1 && *status != cloudformation.StackStatusUpdateComplete && *status != cloudformation.StackStatusCreateComplete && err == nil {
+		select {
+		case <-time.After(periodicCheck):
+			status, reason, err = queryStackStatus(o.svc, o.stackName)
+		}
+	}
+
+	// Multiple status, setup a catch all for all errors.
+	if strings.Index(string(*status), "FAILED") != -1 {
```

Comment:
I managed to get a different state by setting a wrong group name. As this can happen by a wrong user input I suggest we add that here. The state I got was: `ROLLBACK_COMPLETE`. Basically cloudformation failed in the middle of the deployment and decided to roll back and delete everything. I'm not sure if other states could cause issues too, the full list of states can be found here: https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_Stack.html

Reply:
I have fixed that in https://github.com/elastic/beats/pull/8713